internal/impl/snowflake/streaming/int128/int128_test.go

```diff
@@ -0,0 +1,389 @@
+			Neg(Int128{0xf01f1, 0xbc0368bf9a77eae8}),
+		},
+		{
+			Neg(Int128{0x13f837b409a07e7d, 0x7fc8e248a7d73560}),
+			Int64(-0x1b9f),
+			Int128{0xb9157556d724, 0xb14f635714d7563e},
+		},
+	}
+	for _, c := range cases {
+		c := c
+		t.Run("", func(t *testing.T) {
+			require.Equal(
+				t,
+				c.quotient,
+				Div(c.dividend, c.divisor),
+				"%s / %s = %s",
+				c.dividend,
+				c.divisor,
+				c.quotient,
+			)
+		})
+	}
+}
+
+func TestPow10(t *testing.T) {
+	expected := Int64(1)
+	for _, v := range Pow10Table {
+		require.Equal(t, expected, v)
+		expected = Mul(expected, Int64(10))
+	}
+}
+
+func TestCompare(t *testing.T) {
+	tc := [][2]Int128{
+		{Int64(0), Int64(1)},
+		{Int64(-1), Int64(0)},
+		{MinInt128, Int64(0)},
+		{MinInt128, Int64(-1)},
+		{MinInt128, Int64(math.MinInt64)},
+		{MinInt128, Uint64(math.MaxUint64)},
+		{MinInt128, MaxInt128},
+		{Int64(0), MaxInt128},
+		{Int64(-1), MaxInt128},
+		{Int64(math.MinInt64), MaxInt128},
+		{Int64(math.MaxInt64), MaxInt128},
+		{Uint64(math.MaxUint64), MaxInt128},
+	}
+	for _, vals := range tc {
+		a, b := vals[0], vals[1]
+		require.True(t, Less(a, b))
+		require.False(t, Less(b, a))
+		require.True(t, Greater(b, a))
+		require.False(t, Greater(a, b))
+		require.NotEqual(t, a, b)
+		require.Equal(t, a, a)
+		require.Equal(t, b, b)
+	}
+	require.Equal(t, Int64(0), Int64(0))
+	require.NotEqual(t, Int64(1), Int64(0))
+	require.Equal(t, Shl(Int64(1), 64), Add(Uint64(math.MaxUint64), Int64(1)))
+}
+
+func TestParse(t *testing.T) {
+	for _, expected := range [...]Int128{
+		MinInt128,
+		MaxInt128,
+		Int64(0),
+		Int64(-1),
+		Int64(1),
+		MinInt8,
+		MaxInt8,
+		MinInt16,
+		MaxInt16,
+		MinInt32,
+		MaxInt32,
+		MinInt64,
+		MaxInt64,
+		Add(MaxInt64, Uint64(1)),
+	} {
+		actual, ok := Parse(expected.String())
+		require.True(t, ok, "%s", expected)
+		require.Equal(t, expected, actual)
+	}
+	// One less than min
+	_, ok := Parse("-170141183460469231731687303715884105729")
+	require.False(t, ok)
+	// One more than max
+	_, ok = Parse("170141183460469231731687303715884105728")
+	require.False(t, ok)
+}
+
+func TestString(t *testing.T) {
+	require.Equal(t, "-170141183460469231731687303715884105728", MinInt128.String())
+	require.Equal(t, "170141183460469231731687303715884105727", MaxInt128.String())
+}
+
+func TestByteWidth(t *testing.T) {
+	tests := [][2]int64{
+		{0, 1},
+		{1, 1},
+		{-1, 1},
+		{-16, 1},
+		{16, 1},
+		{math.MaxInt8 - 1, 1},
+		{math.MaxInt8, 1},
+		{math.MaxInt8 + 1, 2},
+		{math.MinInt8 - 1, 2},
+		{math.MinInt8, 1},
+		{math.MinInt8 + 1, 1},
+		{math.MaxInt16 - 1, 2},
+		{math.MaxInt16, 2},
+		{math.MaxInt16 + 1, 4},
+		{math.MinInt16 - 1, 4},
+		{math.MinInt16, 2},
+		{math.MinInt16 + 1, 2},
+		{math.MaxInt32 - 1, 4},
+		{math.MaxInt32, 4},
+		{math.MaxInt32 + 1, 8},
+		{math.MinInt32 - 1, 8},
+		{math.MinInt32, 4},
+		{math.MinInt32 + 1, 4},
+		{math.MaxInt64 - 1, 8},
+		{math.MaxInt64, 8},
+		// {math.MaxInt64 + 1, 8},
+		// {math.MinInt64 - 1, 8},
```

Comment:
Do these tests not pass? Might be worth leaving a comment.

Reply:
They overflow :)